ipykernel/zmqshell.py

```diff
@@ -33,7 +33,7 @@
 from IPython.core.magics import MacroToEdit, CodeMagics
 from IPython.core.magic import magics_class, line_magic, Magics
 from IPython.core import payloadpage
-from IPython.core.usage import default_gui_banner
+from IPython.core.usage import default_banner
```

Comment:
The try/except is not necessary, because `default_banner` was already defined. The issue is that the IPython release that removes `default_gui_banner` will break compatibility with any ipykernel releases *before* this PR is merged, including at least ipykernel 4.2.2, the current release. So I don't think we should backport that change to a 4.x release.

Why would it be a problem to have a couple of unused variables sitting in IPython, anyway?